lib/color_guard/middleware.rb

```diff
@@ -11,15 +12,37 @@ def initialize(app, config = {})
     end
 
     def call(env)
-      Thread.current[:color_guard_store] = store(Rack::Request.new(env))
-      response = @app.call(env)
-      Thread.current[:color_guard_store] = nil
+      @req = Rack::Request.new(env)
+      Thread.current[:color_guard_store] = store(@req)
 
-      response
+      if @req.path_info =~ /color_guard\/write_feature/
+        write_feature
+      elsif @req.path_info =~ /color_guard/
+        flags_list
+      else
+        response = @app.call(env)
+        Thread.current[:color_guard_store] = nil
+        response
+      end
     end
 
     private
 
+    def flags_list
+      @all_features = ColorGuard.all
+      view = File.read(File.expand_path("../views/all_features.html.erb", __FILE__))
```

Comment:
No reason to read this file from the filesystem more than once. I'd cache it in memory